src/modules/social/posts/posts.route.ts

```diff
@@ -0,0 +1,130 @@
+import { FastifyInstance } from "fastify"
+
+import { $ref } from "./posts.schema"
+import {
+  getPostsHandler,
+  getPostHandler,
+  createPostHandler,
+  updatePostHandler,
+  reactionHandler
+} from "./posts.controller"
+
+async function postsRoutes(server: FastifyInstance) {
+  server.get(
+    "/",
+    {
+      preHandler: [server.authenticate],
+      schema: {
+        tags: ["posts"],
+        security: [{ bearerAuth: [] }],
+        response: {
+          200: {
+            type: "array",
+            items: $ref("displayPostSchema")
+          }
+        }
+      }
+    },
+    getPostsHandler
+  )
+
+  server.post(
+    "/",
+    {
+      preHandler: [server.authenticate],
+      schema: {
+        body: $ref("createPostBaseSchema"),
+        tags: ["posts"],
+        security: [{ bearerAuth: [] }],
+        response: {
+          200: $ref("postSchema")
+        }
+      }
+    },
+    createPostHandler
+  )
+
+  server.put(
+    "/:id",
+    {
+      preHandler: [server.authenticate],
+      schema: {
+        body: $ref("createPostSchema"),
```

Comment:
We are missing the required id param in the schema here.

```suggestion
      schema: {
        params: {
          type: "object",
          properties: {
            id: { type: "integer" }
          }
        },
        body: $ref("createPostSchema"),
```